app/helpers/application_helper.rb

```diff
@@ -372,4 +372,32 @@ def creator_settings_form?
     creator = User.with_role(:creator).first
     !creator.checked_code_of_conduct && !creator.checked_terms_and_conditions
   end
+
+  # This function is responsible for adding a policy class (and possibly the hidden class).  It's
+  # applying some shortcuts to reduce the likelihood of any Cumulative Layout Shift.
+  #
+  # @param name [String,Symbol] the HTML element name (e.g. "li", "div", "a")
+  # @param record [Object] the record for which we're testing a policy
+  # @param query [Symbol, String] the query we're running on the policy
+  # @param feature_flag [Symbol] the named feature flag that when enabled will add the "hidden"
+  #        class to the dom element.
+  # @param kwargs [Hash] The arguments pass, with modifications to the given :class (see
+  #        implementation details).
+  #
+  # @yield the body of the HTML element
+  #
+  # @see ApplicationPolicy.dom_class_for
+  # @see https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-content_tag
+  # @see ./app/javascript/packs/applyApplicationPolicyToggles.js
+  #
+  # @note This assumes that for the given feature flag, when enabled, we will add the "hidden"
+  #       class.
+  # @note I hope you are correctly busting the cache for this feature_flag
+  def application_policy_content_tag(name, record:, query:, feature_flag:, **kwargs, &block)
+    dom_class = kwargs.delete(:class) || kwargs.delete("class") || ""
```

Comment:
any chance we could leverage `kwargs.with_indifferent_access.delete("class")` here instead of checking for symbol and string?

Reply:
Hmm, I don't usually consider that method.  It partially improves legibility.

I was curious about any performance, so wrote up `benchmarked.rb`.

```ruby
Benchmark.bmbm do |b|
  b.report("with indifferent access via string") do
    1000.times do
      hash = { class: "hello", pancakes: "other"}
      hash.with_indifferent_access.delete("class")
    end
  end
  b.report("with indifferent access via symbol") do
    1000.times do
      hash = { class: "hello", pancakes: "other"}
      hash.with_indifferent_access.delete(:class)
    end
  end
  b.report("with symbol hit access") do
    1000.times do
      hash = { class: "hello", pancakes: "other"}
      hash.delete(:class) || hash.delete("class")
    end
  end
  b.report("with symbol miss but string hit access") do
    1000.times do
      hash = { "class" => "hello", pancakes: "other"}
      hash.delete(:class) || hash.delete("class")
    end
  end
  b.report("with symbol and string miss") do
    1000.times do
      hash = { "no-class" => "hello", pancakes: "other"}
      hash.delete(:class) || hash.delete("class")
    end
  end
end
```

I ran the above with:

`$ bin/rails runner benchmarked.rb`

```sh
Rehearsal --------------------------------------------------------------------------
with indifferent access via string       0.001034   0.000054   0.001088 (  0.001082)
with indifferent access via symbol       0.001033   0.000094   0.001127 (  0.001038)
with symbol hit access                   0.000169   0.000059   0.000228 (  0.000217)
with symbol miss but string hit access   0.000198   0.000070   0.000268 (  0.000245)
with symbol and string miss              0.000166   0.000045   0.000211 (  0.000203)
----------------------------------------------------------------- total: 0.002922sec

                                             user     system      total        real
with indifferent access via string       0.000967   0.000016   0.000983 (  0.000976)
with indifferent access via symbol       0.000908   0.000011   0.000919 (  0.000909)
with symbol hit access                   0.000178   0.000025   0.000203 (  0.000190)
with symbol miss but string hit access   0.000153   0.000014   0.000167 (  0.000154)
with symbol and string miss              0.000189   0.000067   0.000256 (  0.000237)
```

My inclination is to keep the `||` instead of crafting a new object.  We'd also need to consider passing the new object as the `kwargs`.

Reply:
makes sense! appreciate the "extra mile" with benchmarking 💪 FWIW, it wasn't a blocking comment, but more curious if the alternative was considered.

Reply:
I didn't receive it as a blocking comment, but as a chance to ensure we've got the best approach.